OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -228,12 +223,43 @@ bool HandleEvent(ProductionIcon icon, bool isLeftClick, bool handleMultiple)
 					}
 					else
 					{
+						// Pause an existing item
 						Sound.PlayNotification(World.Map.Rules, World.LocalPlayer, "Speech", CurrentQueue.Info.OnHoldAudio, World.LocalPlayer.Country.Race);
 						World.IssueOrder(Order.PauseProduction(CurrentQueue.Actor, icon.Name, true));
 					}
-				}
-				else
+
+					break;
+
+				case 3:
+					// Left click and first != null
+					if (PickUpCompletedBuildingIcon(icon))
+						break;
+					if (first.Paused)
+					{
+						// Resume a paused item
+						Sound.Play(TabClick);
+						World.IssueOrder(Order.PauseProduction(CurrentQueue.Actor, icon.Name, false));
+						break;
+					}
+
+					// falltrough
```

Comment:
It is mostly a matter of opinion. Actually all of the comments seem somewhat redundant, no?